streamer/src/nonblocking/quic.rs

```diff
@@ -130,17 +130,28 @@ pub fn spawn_server(
     max_streams_per_ms: u64,
     wait_for_chunk_timeout: Duration,
     coalesce: Duration,
-) -> Result<(Endpoint, Arc<StreamStats>, JoinHandle<()>), QuicServerError> {
+) -> Result<
+    (
+        Endpoint,
+        Arc<StreamStats>,
+        JoinHandle<()>,
+        u32, /* max_concurrent_connections */
+    ),
+    QuicServerError,
+> {
     info!("Start {name} quic server on {sock:?}");
-    let (config, _cert) = configure_server(keypair)?;
+    let concurrent_connections = max_staked_connections + max_unstaked_connections;
+    let max_concurrent_connections = (concurrent_connections + concurrent_connections / 4) as u32;
+    let (config, _cert) = configure_server(keypair, max_concurrent_connections)?;
 
     let endpoint = Endpoint::new(
         EndpointConfig::default(),
         Some(config),
         sock,
         Arc::new(TokioRuntime),
     )
-    .map_err(QuicServerError::EndpointFailed)?;
+    .unwrap();
```

Comment:
why `unwrap` here as opposed to passing it up the call stack as in the old code?

Reply:
sorry -- unintended changes from forward port from v1.17 based fixes. will revert